services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_insights/sidebar.tsx

```diff
@@ -16,9 +16,22 @@ const Sidebar = ({
   selectedClassroomId,
   setSelectedClassroom
 }) => {
+  const navigate = useNavigate();
+
+  React.useEffect(() => {
```

Comment:
Style/opinion thing: I find it surprising that a component called `Sidebar` has the ability to mutate the URL upon rendering, absent a user interaction

Reply:
I'm not sure I understand -- would it be clearer if I renamed this component `Navigation`, since that's effectively what it is (a sidebar with links)?

Reply:
I don't think renaming the component would make the behavior less surprising. It's just something I haven't encountered before